code/game/machinery/OpTable.dm

```diff
@@ -50,9 +55,21 @@
 		visible_message("<span class='danger'>\The [usr] destroys \the [src]!</span>")
 		src.density = 0
 		qdel(src)
-	return
+		return
+
+	if(!victim)
+		to_chat(user, "<span class='warning'>There is nobody on \the [src]. It would be pointless to turn the suppressor on.</span>")
+		return TRUE
 
+	if(user != victim && !suppressing) // Skip checks if you're doing it to yourself or turning it off, this is an anti-griefing mechanic more than anything.
+		user.visible_message("<span class='warning'>\The [user] begins switching on \the [src]'s neural suppressor.</span>")
+		if(!do_after(user, 30, src) || !user || !src || user.incapacitated() || !user.Adjacent(src))

```

Comment:
`!use_check_and_message(user)` instead of all of that garbage in the ORs.

Reply:
solved